pkg/cmd/cli/cmd/volume.go

```diff
@@ -356,25 +423,30 @@ func (v *VolumeOptions) RunVolume(args []string) error {
 		if err != nil {
 			return err
 		}
-		return p.PrintObj(objects, v.Writer)
+		return p.PrintObj(objects, v.Out)
 	}
 
 	failed := false
 	for _, info := range updateInfos {
 		data, err := info.Mapping.Codec.Encode(info.Object)
 		if err != nil {
-			fmt.Fprintf(v.Writer, "error: %v\n", err)
+			fmt.Fprintf(v.Out, "error: %v\n", err)
 			failed = true
 			continue
 		}
-		obj, err := resource.NewHelper(info.Client, info.Mapping).Replace(info.Namespace, info.Name, true, data)
+		var obj runtime.Object
+		if len(info.ResourceVersion) == 0 {
+			obj, err = resource.NewHelper(info.Client, info.Mapping).Create(info.Namespace, false, data)
+		} else {
+			obj, err = resource.NewHelper(info.Client, info.Mapping).Replace(info.Namespace, info.Name, true, data)
+		}
 		if err != nil {
-			handlePodUpdateError(v.Writer, err, "volume")
+			handlePodUpdateError(v.Out, err, "volume")
```

Comment:
we are printing errors in handlePodUpdateError(), prefer passing v.Err instead of v.Out